src/framework/ui/popover/popover.component.tsx

```diff
@@ -148,15 +148,30 @@ export class PopoverComponent extends React.Component<PopoverProps> {
   private popoverPlacement: PopoverPlacement;
 
   public componentDidUpdate(prevProps: PopoverProps) {
+    this.handleVisibility(prevProps);
+  }
+
+  private handleVisibility = (prevProps: PopoverProps): void => {
     if (prevProps.visible !== this.props.visible) {
       if (this.props.visible) {
         // Toggles re-measuring. This is needed for dynamic containers like ScrollView
         this.setState({ layout: undefined });
       } else {
         this.popoverId = ModalService.hide(this.popoverId);
       }
+    } else if (prevProps.visible && this.props.visible) {
+      this.updatePopoverElement();
```

Comment:
This should be handled in a modal implementation

Reply:
Definitely not. Please read the code

Reply:
I'm not agree with this solution in case it's not. This forces usets to know about this method any time Modal is used